src/media/shortcodes/class-wpml-page-builders-media-shortcodes-update.php

```diff
@@ -50,6 +50,6 @@ public function translate( $post ) {
 			'post_content' => $post->post_content,
 			'tags_input'   => $tag_ids,
 		);
-		wp_update_post( $postarr );
+		wp_update_post( wp_slash( $postarr ) );
```

Comment:
@strategio WordPress already calls `wp_slash` inside `wp_update_post` as long as you pass an object.
You even wrote that in the ticket (*).
So why not passing an object, instead of calling an extra function which, given the right input, will be called by WP anyway?

It's as simple as casing the array with `(object) $postarr`.

(*) I had to jump to the ticket to understand the reason of this change: commit messages are the right place for explanations.

Reply:
@andreasciamanna, I feel that wrapping the array with `wp_slash` is more explicit. If we just cast the array to an object, we don't really have a clue why we are doing this. The reason is hidden inside `wp_update_post`.

Reply:
I guess this is subjective. Looking at the code in the future, I'll still wonder why is that call needed.

Anyway, can you at least fix the commit message?

Other than the ticket title and ID, it contains zero information.

Also, the fact you need to do the same hack multiple times, creates duplication.
I think it would be better if you create a helper function and call this one instead of `wp_uodate_post`.
This helper function can then take care of calling `wp_slash` as well.

Moreover, you can write a specific test for the helper function, while the existing tests can simply check that the helper function is called with the right arguments.

Reply:
@andreasciamanna, I updated the commit message.

Reply:
@strategio not sure why you've marked the conversation as resolved.

What about the code duplication?

Imagine we find out that `wp_slash` is not enough anymore. We would need to find all the places where we are using it and change them, with a high risk to miss some of them.

A helper function would prevent this.

Reply:
@andreasciamanna, you said "at least fix the commit message" so I thought the helper was optional. Honestly, I am divided about this. We already have 2 helpers in Core: `WPML_Create_Post_Helper` and `WPML_WP_Post` which are both doing more or less the same thing (. Sometimes, we use it, sometimes not. I have the feeling that adding one more helper is overkill. And anyway, now we need to know to use `wp_slash` before `wp_update_post`. If we have the helper, all devs will need to know that it is preferable than a direct call to `wp_update_post`.

Can you confirm your idea?

````php
function wpml_update_post_array( array $post_array, $wp_error ) {
    return wp_update_post( wp_slash( $post_array, $wp_error ) );
}
```

Reply:
The ”at least” was for the solution found, since I'm not thrilled about it :)

As for the helper function, I've explained the reasons why it's needed. Code duplication is a dangerous risk I wouldn't personally take.
If you think it's not a valid reason, feel free to dismiss my request.

Reply:
I did some changes as discussed @andreasciamanna, please check again.